draft-handrews-jri.md

```diff
@@ -0,0 +1,536 @@
+
+The full and rather complex history of these keywords, including their usage in OpenAPI and AsyncAPI, as well as other proposals to split some or all of them back into a separate specification, is detailed in the [BACKGROUND.md](https://github.com/json-schema-org/referencing/tree/main/BACKGROUND.md) file in the repository containing this document.
+
+Use cases have been compiled from popular tools that support JSON Reference (and sometimes other keywords in this specification) without being full JSON Schema implementations, and from specification requirements regarding non-JSON Schema use.  Notably, some of the tools support additional JRI keywords that were not included in JSON Reference, motivating a larger specification than simply a revival of JSON Reference.  These requirements and use cases are also available in the [BACKGROUND.md](https://github.com/json-schema-org/referencing/tree/main/BACKGROUND.md) file.
+
+Discussion of this draft takes place in the [GitHub repository](https://github.com/json-schema-org/referencing) and on the [JSON Schema Slack server](https://json-schema.org/slack).
+
+--- middle
+
+# Introduction
+
+JSON Reference and Identification (JRI) offers a modular set of features to enable interoperable support of identification and referencing across a wide range of formats.  These features are necessary to create data formats across a set of linked documents, but are often incidental to the primary functionality of a data format specification.  Offloading these features to common libraries and tools based on a standardized approach allows data format implementations to focus on the format's purpose with minimal need to understand complex and potentially multi-document structures.
+
+Standalone JRI use cases include a variety of document transformations, transparent in-memory representations of linked documents, and locating, caching, and serving resources identified within or referenced from a set of documents.
+
+While {{?RFC8288}} web linking solves this problem in a more general way, a simplified, concise, and self-contained linking and identification system is better-suited to many specifications.  This allows those specifications to avoid complexity, and to avoid constraining their syntax to match any one of the many competing web linking-enabled JSON formats.
+
+To support a variety of use cases, JRI supports trade-offs between generic interoperability and tight integration with the context in which it is used.  JRI defines an interoperable subset of its features for standalone use, while encouraging specifications that need reference and/or identification features to incorporate as much or as little of JRI as suits their needs.  JRI also defines mechanisms for resources to indicate their JRI usage, allowing generic or standalone implementations to understand how to safely process a document without knowledge of its underlying data format.
```

Comment:
Maybe having a statement like, "A context specification MUST NOT include behaviors that conflict with JRI behaviors even if the context specification elects not to utilize those JRI behaviors."  This would prevent a context spec from including `$ref` but defining its own `$defs`.

Reply:
This is addressed in [§4.4 Incorporating a Subset of JRI](https://github.com/json-schema-org/referencing/pull/2/files/8bc3160e0eb19907a85d5bba3ce0ae401ba5f87c#diff-a52f2848d61740c55c6f7756031d478fc004b2e4a369b6456e5eec2a1563988dR407), which was originally much earlier in the document.

I'll have to figure out where the best home for that is.  I don't want to overwhelm with too much context specification-related information before defining the basics, but this should probably be mentioned early on.  Perhaps even tacking on a "as long as they do not redefine the unused portion" or something similar in this paragraph (the intro paragraph shouldn't have any RFC 2119 keywords, IMHO, as it's just a casual overview).